src/omnisharp/server.ts

```diff
@@ -445,22 +448,39 @@ export class OmniSharpServer {
                 });
             }
 
-            // If there's more than one launch target, we start the server if one of the targets
-            // matches the preferred path. Otherwise, we fire the "MultipleLaunchTargets" event,
-            // which is handled in status.ts to display the launch target selector.
-            if (launchTargets.length > 1 && preferredPath) {
+            this._options = Options.Read(this.vscode);
 
-                for (let launchTarget of launchTargets) {
-                    if (launchTarget.target === preferredPath) {
-                        // start preferred path
-                        return this.restart(launchTarget);
-                    }
+            const defaultLaunchSolutionConfigValue = this._options.defaultLaunchSolution;
+
+            // does the configured target actually exist
+            const defaultLaunchSolutionTarget = launchTargets.filter((a) => (path.basename(a.target) === defaultLaunchSolutionConfigValue));
```

Comment:
Since you're only interested in the first match, Array.Find might be a little clearer (it just returns the element or `undefined`) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
I think if you do this you can just do the default launch target match check before the if block that starts on line 451 and not touch that block at all?